core/src/main/java/org/elasticsearch/search/slice/SliceBuilder.java

```diff
@@ -194,9 +196,14 @@ public Query toFilter(QueryShardContext context, int shardId, int numShards) {
             throw new IllegalArgumentException("field " + field + " not found");
         }
 
+        String field = this.field;
         boolean useTermQuery = false;
         if (UidFieldMapper.NAME.equals(field)) {
-           useTermQuery = true;
+            if (context.getIndexSettings().isSingleType()) {
```

Comment:
slicing is not about scoring, it's about partitioning the data afaik? with lucene doc IDs you can not resume a slice since you might hit a different replica when you resume. I wonder if that is a different problem here... for random score I agree we might just go down that path by default and if somebody needs reproducibility we can still use something like the ID or seq ids?